astropy/io/misc/yaml.py

```diff
@@ -222,6 +223,19 @@ def ignore_aliases(self, data):
 AstropyDumper.add_representer(Time, _time_representer)
 AstropyDumper.add_representer(TimeDelta, _timedelta_representer)
 AstropyDumper.add_representer(coords.SkyCoord, _skycoord_representer)
+AstropyDumper.add_representer(np.int32, \
+                              yaml.representer.Representer.represent_int)
+try:
+    AstropyDumper.add_representer(np.int64, \
+                                  yaml.representer.Representer.represent_long)
+except:
+    warnings.warn("This version of Pyyaml does not support long ints; using "
```

Comment:
That's true only for `SafeRepresenter`. In `Representer` they are different:
https://github.com/yaml/pyyaml/blob/master/lib/yaml/representer.py#L320